base/errorshow.jl

```diff
@@ -602,6 +601,11 @@ function show_backtrace(io::IO, t::Vector{Any})
     end
 end
 
+function is_kw_sorter_name(name::Symbol)
+    sn = string(name)
+    return !startswith(sn, '#') && endswith(sn, "##kw")
```

Comment:
Is this a problem?
```julia
julia> module Toplevel

       fname = Symbol("#hidden")
       @eval ($fname)(x; y=1, z="hi") = 1

       end

julia> names(Toplevel, all=true)
11-element Array{Symbol,1}:
 Symbol("###hidden#1") 
 Symbol("##hidden")    
 Symbol("##hidden#1")  
 Symbol("#eval")       
 Symbol("#hidden")     
 Symbol("#include")    
 Symbol("#kw###hidden")
 :Toplevel             
 :eval                 
 :fname                
 :include              
```

Reply:
Could be, but there are probably worse collisions that can happen if user identifiers contain `#`, so unofficially that is just not supported.

Reply:
@timholy we now have this syntax for your example :grimacing:

```julia
module Toplevel
var"#hidden"(x; y=1, z="hi") = 1
end
```

Reply:
Cool, hadn't followed that. :+1:

Reply:
:fearful:

Reply:
@JeffBezanson indeed... unintended consequences.

Reply:
It's still worth having `var` I suspect. As this reveals, making important decisions on the basis of naming has some potential downsides. One could potentially solve this and #30908 by introducing more fields linking the various lowered representation to one another. Then you wouldn't be beholden to a naming convention.

Reply:
Agreed. I mean, we knew `var` *could* be used for this kind of thing... just I'd originally envisaged it as more of an internal thing or occasional escape hatch for naming variables in FFIs. Seeing it becoming directly "useful" is kind of disturbing. The most surprising use so far has been the DataFrames people using it to refer to columns with poorly normalized names.